sdc11073/wsdiscovery.py

```diff
@@ -820,45 +831,42 @@ def removeSourceAddr(self, addr):
     def addUnicastMessage(self, env, addr, port, initialDelay=0):
         msg = Message(env, addr, port, Message.UNICAST, initialDelay)
         self._logger.debug('addUnicastMessage: adding message Id %s. delay=%.2f'.format(env.getMessageId(), initialDelay))
-        self._queue.put(msg)
+        self._repeated_enqueue_msg(msg, initialDelay, UNICAST_UDP_REPEAT, UNICAST_UDP_MIN_DELAY,
+                                   UNICAST_UDP_MAX_DELAY, UNICAST_UDP_UPPER_DELAY)
 
     def addMulticastMessage(self, env, addr, port, initialDelay=0):
         msg = Message(env, addr, port, Message.MULTICAST, initialDelay)
         self._logger.debug('addMulticastMessage: adding message Id %s. delay=%.2f'.format(env.getMessageId(), initialDelay))
-        self._queue.put(msg)
+        self._repeated_enqueue_msg(msg, initialDelay, MULTICAST_UDP_REPEAT, MULTICAST_UDP_MIN_DELAY,
+                                   MULTICAST_UDP_MAX_DELAY, MULTICAST_UDP_UPPER_DELAY)
+
+    def _repeated_enqueue_msg(self, msg, initial_delay_ms, repeat, min_delay_ms, max_delay_ms, upper_delay_ms):
+        next_send = time.time() + initial_delay_ms/1000.0
+        dt = (min_delay_ms + ((max_delay_ms - min_delay_ms) * random.random())) / 2000.0

```

Comment:
Not a fan of the magic value here. Also, consider using ` random.randrange(start, stop[, step])` to express a random integer within a range